src/main/java/de/tum/in/www1/artemis/config/Constants.java

```diff
@@ -213,6 +213,16 @@ public final class Constants {
 
     public static final String HAZELCAST_EXERCISE_CACHE = HAZELCAST_QUIZ_PREFIX + "exercise-cache";
 
+    public static final long MONITORING_SAVE_DELAY = 60 * 30 * 1000; // 30 minutes in milliseconds
```

Comment:
Just an idea: Would it make sense to make this configurable in the future or is 30 minutes the sweet spot?